src/applications/vaos/services/patient/index.js

```diff
@@ -0,0 +1,388 @@
+    ).catch(createErrorHandler('request-check-past-visits-error'));
+  }
+
+  if (type !== 'request' && typeOfCare.id !== PRIMARY_CARE) {
+    checks.directPastVisit = checkPastVisits(
+      location.vistaId,
+      location.id,
+      typeOfCare.id,
+      'direct',
+    ).catch(createErrorHandler('direct-check-past-visits-error'));
+  }
+
+  const results = await promiseAllFromObject(checks);
+  const output = { direct: null, request: null };
+
+  if (results.directPastVisit instanceof Error) {
+    output.direct = new Error('Direct scheduling eligibility check error');
+  } else if (type !== 'request') {
+    output.direct = {
+      hasRequiredAppointmentHistory: hasVisitedInPastMonths(
+        results.directPastVisit,
+      ),
+    };
+  }
+
+  if (
+    results.requestPastVisit instanceof Error ||
+    results.requestLimit instanceof Error
+  ) {
+    output.request = new Error('Request eligibility check error');
+  } else if (type !== 'direct') {
+    output.request = {
+      hasRequiredAppointmentHistory: hasVisitedInPastMonths(
+        results.requestPastVisit,
+      ),
+      isEligibleForNewAppointmentRequest: isUnderRequestLimit(
+        results.requestLimit,
+      ),
+    };
+  }
+
+  return output;
+}
+
+/**
+ * Returns patient based eligibility checks for specified request or direct types
+ *
+ * @export
+ * @param {Object} params
+ * @param {TypeOfCare} params.typeOfCare Type of care object,
+ * @param {Location} params.location Location of where patient should have eligibility checked,
+ * @param {'direct'|'request'|null} [params.type=null] The type to check eligibility for. By default,
+ *   will check both
+ * }
+ * @returns {PatientEligibility} Patient eligibility data
+ */
+export async function fetchPatientEligibility({
+  typeOfCare,
+  location,
+  type = null,
+}) {
+  return fetchPatientEligibilityFromVAR({ typeOfCare, location, type });
+}
+
+function locationSupportsDirectScheduling(location, typeOfCare) {
+  return (
+    // this check is included due to old two step facilities page
+    location.legacyVAR.directSchedulingSupported ||
+    location.legacyVAR.settings?.[typeOfCare.id]?.direct.enabled
+  );
+}
+
+function locationSupportsRequests(location, typeOfCare) {
+  return (
+    // this check is included due to old two step facilities page
+    location.legacyVAR.requestSupported ||
+    location.legacyVAR.settings?.[typeOfCare.id]?.request.enabled
+  );
+}
+
+function hasMatchingClinics(clinics, pastAppointments) {
+  return clinics?.some(
+    clinic =>
+      !!pastAppointments.find(appt => {
+        return (
+          clinic.identifier[0].value ===
+          `urn:va:healthcareservice:${appt.facilityId}:${appt.sta6aid}:${
+            appt.clinicId
+          }`
+        );
+      }),
+  );
+}
+
+/*
+ * This function logs information about eligibility to the console, to help with
+ * testing in non-production environments
+ */
+/* istanbul ignore next */
+function logEligibilityExplanation(
```

Comment:
Same idea as the function in eligibility.js, but simpler because we can just map things from the reason codes.